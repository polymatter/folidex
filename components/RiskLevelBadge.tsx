import React from 'react'
import { StyleSheet } from 'react-native';
import { Text } from '../components/Themed';
import { MaterialIcons, MaterialCommunityIcons } from '@expo/vector-icons';
import Colors from '../constants/Colors';
import { RiskLevel } from '../types'

const HighRed = Colors.badgeLevels.High;
const MediumYellow = Colors.badgeLevels.Medium;
const LowGreen = Colors.badgeLevels.Low;

const RiskLevelBadge = ({ level }: { level: RiskLevel }) => {
  switch (level) {
    case 'High':
      return <>
        <MaterialIcons name="dangerous" size={24} color={HighRed} />
        <Text style={[styles.levelBadge, styles.levelBadgeHigh]}>High</Text>
      </>
    case 'Medium':
      return <>
        <MaterialIcons name="warning" size={24} color={MediumYellow} />
        <Text style={[styles.levelBadge, styles.levelBadgeMedium]}>Medium</Text>
      </>
    case 'Low':
      return <>
        <MaterialCommunityIcons name="emoticon-happy-outline" size={24} color={LowGreen} />
        <Text style={[styles.levelBadge, styles.levelBadgeLow]}>Low</Text>
      </>
  }
}

const styles = StyleSheet.create({
  label: {
    display: 'flex',
    flexGrow: 10
  },
  levelBadge: {
    display: 'flex',
    flexShrink: 0,
    borderRadius: 15,
    fontSize: 12,
    paddingLeft: 8,
    paddingRight: 8,
    marginRight: 30,
  },
  levelBadgeLow: {
    color: LowGreen,
  },
  levelBadgeMedium: {
    color: MediumYellow,
  },
  levelBadgeHigh: {
    color: HighRed,
  },
});

export default RiskLevelBadge
