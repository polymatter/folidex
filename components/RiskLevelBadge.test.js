import * as React from 'react';
import { render, screen } from '@testing-library/react-native';
import RiskLevelBadge from './RiskLevelBadge';

jest.mock('@expo/vector-icons', () => ({
  AntDesign: 'Icon',
  Entypo: 'Icon',
  EvilIcons: 'Icon',
  Feather: 'Icon',
  Fontisto: 'Icon',
  FontAwesome: 'Icon',
  FontAwesome5: 'Icon',
  Foundation: 'Icon',
  Ionicons: 'Icon',
  MaterialCommunityIcons: 'Icon',
  MaterialIcons: 'Icon',
  Octicons: 'Icon',
  SimpleLineIcons: 'Icon',
  Zocial: 'Icon',
  createMultiStyleIconSet: 'Icon',
  createIconSet: 'Icon',
  createIconSetFromFontello: 'Icon',
  createIconSetFromIcoMoon: 'Icon',
  createMu: 'Icon'
}));

describe('<RiskLevelBadge>', () => {

  test('can render a high risk badge', () => {

    const tree = render(
      <RiskLevelBadge level={'High'} />
    ).toJSON();

    expect(tree).toMatchSnapshot();
    expect(screen.getByText('High')).toBeInTheDocument();
  })

  test('can render a medium risk badge', () => {

    const tree = render(
      <RiskLevelBadge level={'Medium'} />
    ).toJSON();

    expect(tree).toMatchSnapshot();
    expect(screen.getByText('Medium')).toBeInTheDocument();
  })

  test('can render a low risk badge', () => {

    const tree = render(
      <RiskLevelBadge level={'Low'} />
    ).toJSON();

    expect(tree).toMatchSnapshot();
    expect(screen.getByText('Low')).toBeInTheDocument();
  })
})