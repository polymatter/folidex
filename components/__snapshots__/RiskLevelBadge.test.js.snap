// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RiskLevelBadge> can render a high risk badge 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#fff",
      },
      Object {
        "alignItems": "center",
        "flex": 1,
        "flexDirection": "row",
        "lineHeight": 10,
        "minHeight": 30,
        "paddingLeft": 5,
        "paddingRight": 5,
      },
    ]
  }
>
  <Icon
    color="#dc3545"
    name="dangerous"
    size={24}
  />
  <Text
    style={
      Array [
        Object {
          "color": "#000",
        },
        Array [
          Object {
            "borderRadius": 15,
            "display": "flex",
            "flexShrink": 0,
            "fontSize": 12,
            "marginRight": 30,
            "paddingLeft": 8,
            "paddingRight": 8,
          },
          Object {
            "color": "#dc3545",
          },
        ],
      ]
    }
  >
    High
  </Text>
</View>
`;

exports[`<RiskLevelBadge> can render a low risk badge 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#fff",
      },
      Object {
        "alignItems": "center",
        "flex": 1,
        "flexDirection": "row",
        "lineHeight": 10,
        "minHeight": 30,
        "paddingLeft": 5,
        "paddingRight": 5,
      },
    ]
  }
>
  <Icon
    color="#29c338"
    name="emoticon-happy-outline"
    size={24}
  />
  <Text
    style={
      Array [
        Object {
          "color": "#000",
        },
        Array [
          Object {
            "borderRadius": 15,
            "display": "flex",
            "flexShrink": 0,
            "fontSize": 12,
            "marginRight": 30,
            "paddingLeft": 8,
            "paddingRight": 8,
          },
          Object {
            "color": "#29c338",
          },
        ],
      ]
    }
  >
    Low
  </Text>
</View>
`;

exports[`<RiskLevelBadge> can render a medium risk badge 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#fff",
      },
      Object {
        "alignItems": "center",
        "flex": 1,
        "flexDirection": "row",
        "lineHeight": 10,
        "minHeight": 30,
        "paddingLeft": 5,
        "paddingRight": 5,
      },
    ]
  }
>
  <Icon
    color="#efa73c"
    name="warning"
    size={24}
  />
  <Text
    style={
      Array [
        Object {
          "color": "#000",
        },
        Array [
          Object {
            "borderRadius": 15,
            "display": "flex",
            "flexShrink": 0,
            "fontSize": 12,
            "marginRight": 30,
            "paddingLeft": 8,
            "paddingRight": 8,
          },
          Object {
            "color": "#efa73c",
          },
        ],
      ]
    }
  >
    Medium
  </Text>
</View>
`;
